lib/trip_dispatcher.rb

```diff
@@ -27,14 +34,38 @@ def inspect
               #{passengers.count} passengers>"
     end
 
+    def request_trip(passenger_id)
+      passenger = find_passenger(passenger_id)
+      driver = @drivers.select {|driver| driver.status == :AVAILABLE}.first
+      raise ArgumentError, "There are no available drivers" if driver == nil
+ 
+      new_trip = Trip.new(
+        id: @trips.length + 1,
+        driver: driver,
+        passenger: passenger,
+        start_time: Time.now, 
+        end_time: nil,
+        cost: nil,
+        rating: nil
+    )
+
+      new_trip.connect(passenger,driver)
+
+      driver.update_status
```

Comment:
You can just use the accessor here (see above):

```suggestion
      driver.status = :UNAVAILABLE
```